Dbradham/761 idea space main page responsive

dbradham opened the pull request:
This pull request will address issue #761 and also issues with the /join page and other components throughout the idea space application. (such as Links that are broken on the first click)

I tried to consolidate and clean code, delete as much unused code and as many files as I could, as well as some .css files to use more styled components.

This will also unblock our release pipeline!

dbradham commented on apps/ideaspace/src/components/modules/WelcomePage/StyledWelcomePage.js:
```diff
@@ -1,32 +1,24 @@
-  min-height: 50vh;
-  height: 1px;
-  max-width: 1700px;
+export const ButtonArea = styled.aside`
```
@Enjoy2Live @pyxld-kris I don't really like to use aside, but this was broken unless I did. I found the solution here: https://github.com/vercel/next.js/issues/4068

This seems like kind of a strange workaround-I'm not sure if there is some deeper problem with next/styled-components, but I wanted to call it out in case anyone runs into it in the future.

<img width="953" alt="Screen Shot 2023-02-18 at 8 46 28 PM" src="https://user-images.githubusercontent.com/43216378/219909392-a02c9203-b40e-402c-8b74-72613d6c8404.png">

pyxld-kris replied:
I've never seen this before so I appreciate the callout. It being a problem/solution from back in 2021 is a bit concerning.

Enjoy2Live commented on apps/dev-recruiters/src/components/modules/OpportunitiesAggregatorWithRoles/OpportunitiesAggregator.tsx:
```diff
@@ -4,13 +4,15 @@ import {
   FilteringComponentProps
 } from "./filtering/FilteringComponent";
 import ShortCard from "./ShortCard";
+import '../hide-slick-dots.css';
```
@dbradham 
I think commenting the import statement of `slick-carousel/slick/slick-theme.css` has nothing to do with the app breaking on /join. can please uncomment it and remove line `import '../hide-slick-dots.css';` and its file

dbradham replied:
If I uncomment that line, I see the below errors and the application becomes unresponsive as soon as I click /join. I'm open to other suggestions to solve this, as this is not the most beautiful solution. (but is a needed solution from my testing)

<img width="755" alt="Screen Shot 2023-02-20 at 11 19 55 AM" src="https://user-images.githubusercontent.com/43216378/220168603-d5b0deec-9235-4355-b6a1-58d047e18d3e.png">

Enjoy2Live replied:
@dbradham that's probably an issue with your environment, mind if you send me node version and the os?
on my side everything works as expected without commenting the slick css and it's important for the functionality of the carousel

Enjoy2Live replied:
for me I'm using 
node: v14.17.0
os: Pop!_OS 22.04 LTS

dbradham replied:
@Enjoy2Live Thanks that seems to be the issue. I have been using node v18.12.1 on Monterey OS (12.3). I tried on my windows 11 PC with v16.16.0 Node and it seems to be working as expected. I just updated the PR to revert these changes.

dbradham commented:
@Enjoy2Live @pyxld-kris please let me know if there is anything else to address or if this can be merged.

pyxld-kris commented:
@dbradham 

**I just went through and tested the app and things look good!** There was only one thing...

I noticed that **the AI-powered idea generator on IdeaSpace isn't working**. After doing some digging, it looks like **they've discontinued the API** we were using. You can see their deprecation here: https://www.banana.dev/pretrained-models/nodejs/gptj

**I've made a quick, simple, ugly patch that might be suitable to keep things on track for the release.** Basically, I changed the modal that pops up to display a message that says "This feature is currently offline. We apologize for the inconvenience, please check back soon!". I've attached an image showing what I've done. 

![image](https://user-images.githubusercontent.com/46331884/221287075-dd5e4112-0ca5-4c57-8d5d-c9415fcc0c0b.png)

Short-term, hopefully, this helps keep things on track. Long term you guys will want to figure out how to best tackle remedying this, either by removing the feature completely or patching it. I'm open to using a paid service if it's within a reasonable budget and you guys see value in the feature, or we may be able to establish some relationships with a company that can provide free credits to our nonprofit org!

pyxld-kris commented:
Other than that, this seems mergeable to me.

dbradham commented:
> @dbradham
> 
> **I just went through and tested the app and things look good!** There was only one thing...
> 
> I noticed that **the AI-powered idea generator on IdeaSpace isn't working**. After doing some digging, it looks like **they've discontinued the API** we were using. You can see their deprecation here: https://www.banana.dev/pretrained-models/nodejs/gptj
> 
> **I've made a quick, simple, ugly patch that might be suitable to keep things on track for the release.** Basically, I changed the modal that pops up to display a message that says "This feature is currently offline. We apologize for the inconvenience, please check back soon!". I've attached an image showing what I've done.
> 
> ![image](https://user-images.githubusercontent.com/46331884/221287075-dd5e4112-0ca5-4c57-8d5d-c9415fcc0c0b.png)
> 
> Short-term, hopefully, this helps keep things on track. Long term you guys will want to figure out how to best tackle remedying this, either by removing the feature completely or patching it. I'm open to using a paid service if it's within a reasonable budget and you guys see value in the feature, or we may be able to establish some relationships with a company that can provide free credits to our nonprofit org!



> @dbradham
> 
> **I just went through and tested the app and things look good!** There was only one thing...
> 
> I noticed that **the AI-powered idea generator on IdeaSpace isn't working**. After doing some digging, it looks like **they've discontinued the API** we were using. You can see their deprecation here: https://www.banana.dev/pretrained-models/nodejs/gptj
> 
> **I've made a quick, simple, ugly patch that might be suitable to keep things on track for the release.** Basically, I changed the modal that pops up to display a message that says "This feature is currently offline. We apologize for the inconvenience, please check back soon!". I've attached an image showing what I've done.
> 
> ![image](https://user-images.githubusercontent.com/46331884/221287075-dd5e4112-0ca5-4c57-8d5d-c9415fcc0c0b.png)
> 
> Short-term, hopefully, this helps keep things on track. Long term you guys will want to figure out how to best tackle remedying this, either by removing the feature completely or patching it. I'm open to using a paid service if it's within a reasonable budget and you guys see value in the feature, or we may be able to establish some relationships with a company that can provide free credits to our nonprofit org!

Thanks Kris. I'll plan to discuss this with @JulieMass and the team.

I don't see any obvious free alternatives from doing a quick search, but we may dedicate a research spike to this.

dev-launchers-flux commented:
:tada: This PR is included in version 2.10.0 :tada:

The release is available on [GitHub release](https://github.com/dev-launchers/dev-launchers-platform/releases/tag/v2.10.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
